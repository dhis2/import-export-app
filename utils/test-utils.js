import * as testingLibrary from '@testing-library/react'
import * as customQueries from './custom-queries.js'

const { queries } = testingLibrary

console.log('testingLibrary', testingLibrary)

const customRender = (ui, options) =>
    testingLibrary.render(ui, {
        queries: { ...queries, ...customQueries },
        ...options,
    })

/* eslint-disable */
// re-export everything
export * from '@testing-library/react'

// override render method
export { customRender as render }
/* eslint-enable */
