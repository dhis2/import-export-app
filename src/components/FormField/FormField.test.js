import React from 'react'
import { render } from 'test-utils'
import '@testing-library/jest-dom/extend-expect.js'

import { FormField } from '../index.js'

const props = {
    dataTest: 'format',
    label: 'Format',
}

it('matches snapshot', () => {
    const { asFragment } = render(<FormField {...props} />)
    expect(asFragment()).toMatchSnapshot()
})

it('shows children', () => {
    const r = (
        <FormField {...props}>
            <p>Child paragraph 1</p>
        </FormField>
    )
    const { getByDataTest } = render(r)
    const ff = getByDataTest('format')
    expect(ff).toHaveTextContent('Child paragraph 1')
})

it('shows star when required', () => {
    const r = (
        <FormField {...props} required>
            <p>Child paragraph 1</p>
        </FormField>
    )
    const { getByDataTest } = render(r)
    const ff = getByDataTest('format')
    expect(ff).toContainHTML('<span> *</span>')
})
