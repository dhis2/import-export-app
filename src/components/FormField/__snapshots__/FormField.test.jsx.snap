// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`matches snapshot 1`] = `
<DocumentFragment>
  <div
    class="container"
    data-test="format"
  >
    <span
      class="label"
    >
      Format
    </span>
  </div>
</DocumentFragment>
`;
